build-system/compile/internal-version.js

```diff
@@ -26,40 +26,75 @@ const argv = minimist(process.argv.slice(2), {
+  //   cherry-picked from `master`, then <X> is 0).
+  //   - If <X> > 9 then cap it at 9 (or throw an error for `--strict_build`s).
+  // - Find the commit (<C>) before the last cherry-picked commit from the
+  //   `master` branch (if the current branch is `master`, or otherwise in
+  //   `master`'s commit history, then the current commit is <C>).
+  // - Find the commit time of <C> (<C>.time). Note that commit time might be
+  //   different from author time! e.g., commit time might be the time that a PR
+  //   was merged into `master`, or a commit was cherry-picked onto the brabnch;
+  //   author time is when the original author of the commit ran the
+  //   "git commit" command.
+  // - The version number is <C>.time.format("YYmmDDHHMMSS", "UTC") + <X> (the
+  //   pseudo-code assumes standard `.strftime` formatting).
+  //
+  // Examples:
+  // 1. The version number of a release built from the HEAD commit on `master`,
+  //    where that HEAD commit was committed on April 25, 2020 2:31:11 PM EDT
+  //    would be `2004251831110`.
+  //    - EDT is UTC-4, so the hour value changes from EDT's 14 to UTC's 18.
+  //    - The last digit is 0 as this commit is on `master`.
+  //
+  // 2. The version number of a release built from a local working branch (e.g.,
+  //    on a developer's workstation) that was split off from a `master` commit
+  //    from May 6, 2021 10:40:59 AM PDT and has multiple commits that exist
+  //    only on local working branch would be `2105061840590`.
+  //    - PDT is UTC-7, so the hour value changes from PST's 10 to UTC's 17.
+  //    - The last digit is 0 as this commit is on a branch that was split
+  //      from but does not have any commits since the split that were
+  //      cherry-picked from `master`.
+  //
+  // 3. For a release built from a local working branch that was split off from
+  //    a `master` commit from November 9, 2021 11:48:11 PM PST, and then:
+  //    - had one commit that was cherry-picked from `master`,
+  //    - followed by two commits that were created directly on the branch, the
+  //      last of which was commited on November 10, 2021 5:01:12 PM PST,
+  //    - followed by four commits that were cherry-picked from `master`,
+  //    then its version number would be `202111110101124`.
+  //    - The latest four commits are cherry-picks from `master`, and the one
+  //      before them is not, so our last digit is set to 4.
+  //    - PST is UTC-8, so the hour value changes from PST's 17 to UTC's 1, and
+  //      one day is added.
+  let numberOfCherryPicks = 0;
+  const commitCherriesInfo = gitCherryMaster().reverse();
+  for (const {isCherryPick} of commitCherriesInfo) {
+    if (!isCherryPick) {
+      break;
     }
-    return `${lastCommitFormattedTime}${lastDigit}`;
+    numberOfCherryPicks++;
+  }
+  const lastCommitFormattedTime = gitCommitFormattedTime(
+    `HEAD~${numberOfCherryPicks}`
+  );
+
+  if (argv.strict_build && numberOfCherryPicks > 9) {
```

Comment:
usability nit: `--strict_build` didn't convey what this meant until I read all the comments. How about `--verify_cherry_picks`? (Also, see my proposal below to avoid adding this flag altogether.)

Reply:
Obsolete